RecoCTPPS/PixelLocal/plugins/CTPPSPixelLocalTrackProducer.cc

```diff
@@ -0,0 +1,258 @@
+  if(patterFinderAlgorithm == "RPixRoadFinder"){
+   patternFinder_ = new RPixRoadFinder(parameterSet_);
+  }
+  else{
+    throw cms::Exception("CTPPSPixelLocalTrackProducer") << "Pattern finder algorithm" << patterFinderAlgorithm << " does not exist";
+  }
+  
+  for(uint32_t i=0; i<numberOfPlanesPerPot_; ++i){
+    listOfAllPlanes_.push_back(i);
+  }
+
+  //tracking algorithm selector
+  if(trackFitterAlgorithm == "RPixPlaneCombinatoryTracking"){
+    trackFinder_ = new RPixPlaneCombinatoryTracking(parameterSet_);
+  }
+  else{
+    throw cms::Exception("CTPPSPixelLocalTrackProducer") << "Tracking fitter algorithm" << trackFitterAlgorithm << " does not exist";
+  }
+  trackFinder_->setListOfPlanes(listOfAllPlanes_);
+  trackFinder_->initialize();
+  
+  tokenCTPPSPixelRecHit_ = consumes<edm::DetSetVector<CTPPSPixelRecHit> >(edm::InputTag(inputTag_));
+ 
+  produces<edm::DetSetVector<CTPPSPixelLocalTrack> > ();
+
+
+ 
+}
+
+//------------------------------------------------------------------------------------------------//
+
+CTPPSPixelLocalTrackProducer::~CTPPSPixelLocalTrackProducer(){
+  delete patternFinder_;
+  delete trackFinder_ ;
+}
+
+//------------------------------------------------------------------------------------------------//
+
+void CTPPSPixelLocalTrackProducer::fillDescriptions(edm::ConfigurationDescriptions & descriptions){
+  edm::ParameterSetDescription desc;
+
+  desc.add<std::string>    ("label"                            , "ctppsPixelRecHits"           );
+  desc.add<std::string>    ("RPixPatterFinderAlgorithm"        , "RPixRoadFinder"              );
+  desc.add<std::string>    ("RPixTrackFinderAlgorithm"         , "RPixPlaneCombinatoryTracking");
+  desc.addUntracked<uint>  ("RPixTrackMinNumberOfPoints"       , 3                             );
+  desc.addUntracked<int>   ("RPixVerbosity"                    , 0                             );
+  desc.add<double>         ("MaximumChi2OverNDF"               , 5.                            );
+  desc.add<double>         ("MaximumXLocalDistanceFromTrack"   , 0.2                           );
+  desc.add<double>         ("MaximumYLocalDistanceFromTrack"   , 0.3                           );
+  desc.addUntracked<int>   ("RPixMaxHitPerPlane"               , 20                            );
+  desc.addUntracked<int>   ("RPixMaxHitPerRomanPot"            , 60                            );
+  desc.addUntracked<int>   ("RPixMaxTrackPerRomanPot"          , 10                            );
+  desc.addUntracked<int>   ("RPixMaxTrackPerPattern"           , 5                             );
+  desc.addUntracked<int>   ("NumberOfPlanesPerPot"             , 6                             );
+  desc.add<double>         ("RPixRoadRadius"                   , 1.0                           );
+  desc.add<int>            ("RPixMinRoadSize"                  , 3                             );
+  desc.add<int>            ("RPixMaxRoadSize"                  , 20                            );
```

Comment:
documentation text can be added to the parameters to help in possible interactive use cases